src/dex/balancer-v1/balancer-v1.ts

```diff
@@ -0,0 +1,729 @@
+
+  readonly exchangeRouterInterface: Interface;
+
+  readonly exchangeProxy: Address =
+    '0x6317c5e82a06e1d8bf200d21f4510ac2c038ac81';
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(BalancerV1Config);
+
+  logger: Logger;
+
+  constructor(
+    protected network: Network,
+    protected dexKey: string,
+    protected dexHelper: IDexHelper,
+    protected subgraphURL: string = BalancerV1Config[dexKey] &&
+      BalancerV1Config[dexKey][network].subgraphURL,
+    protected adapters = Adapters[network] || {},
+  ) {
+    super(dexHelper.augustusAddress, dexHelper.provider);
+    this.logger = dexHelper.getLogger(dexKey);
+    this.eventPools = new BalancerV1EventPool(
+      dexKey,
+      network,
+      dexHelper,
+      this.logger,
+      adapters,
+    );
+    this.exchangeRouterInterface = new Interface(BalancerV1ABI);
+  }
+
+  async setupEventPools(blockNumber: number) {
+    const poolState = await this.eventPools.generateState(blockNumber);
+    this.eventPools.setState(poolState, blockNumber);
+    this.dexHelper.blockManager.subscribeToLogs(
+      this.eventPools,
+      this.eventPools.addressesSubscribed,
+      blockNumber,
+    );
+  }
+
+  async initializePricing(blockNumber: number) {
+    await this.setupEventPools(blockNumber);
+  }
+
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return this.adapters[side] || null;
+  }
+
+  static getIdentifier(dexKey: string, address: string) {
+    return `${dexKey}_${address.toLowerCase()}`;
+  }
+
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    if (srcToken.address.toLowerCase() === destToken.address.toLowerCase())
+      return [];
+
+    const _from = wrapETH(srcToken, this.network);
+    const _to = wrapETH(destToken, this.network);
+
+    const pools = this.getPools(_from, _to);
+
+    return pools.map(({ id }) => BalancerV1.getIdentifier(this.dexKey, id));
+  }
+
+  async updatePoolState(
+    pools?: PoolState[],
+    blockNumber?: number,
+  ): Promise<void> {
+    const _pools = pools === undefined ? this.eventPools.allPools : pools;
+
+    if (_pools.length === 0) throw Error('There are no pools.');
+
+    let addresses: string[][] = [];
+    let total = 0;
+
+    for (let i = 0; i < _pools.length; i++) {
+      let pool = _pools[i];
+
+      addresses.push([pool.id]);
+      total++;
+      pool.tokens.forEach(token => {
+        addresses[i].push(token.address);
+        total++;
+      });
+    }
+
+    let results = await this.eventPools.balancerMulticall.methods
+      .getPoolInfo(addresses, total)
+      .call({}, blockNumber);
+
+    let j = 0;
+    for (let i = 0; i < _pools.length; i++) {
+      _pools[i].tokens.forEach(token => {
+        token.balance = bignumberify(bmath.bnum(results[j]));
+        j++;
+      });
+    }
+  }
```

Comment:
The function is not used anywhere?

Reply:
Removed. Not need in this implementation